`make install` no longer working in `sdk/nodejs`

I'm now seeing this:

```
➜  nodejs git:(master) ✗ make install
INSTALL:
[ ! -e "/Users/lukehoban/.pulumi-dev/node_modules/@pulumi/pulumi" ] || rm -rf "/Users/lukehoban/.pulumi-dev/node_modules/@pulumi/pulumi"
mkdir -p "/Users/lukehoban/.pulumi-dev/node_modules/@pulumi/pulumi"
cp -r bin/. "/Users/lukehoban/.pulumi-dev/node_modules/@pulumi/pulumi"
cp yarn.lock "/Users/lukehoban/.pulumi-dev/node_modules/@pulumi/pulumi"
rm -rf "/Users/lukehoban/.pulumi-dev/node_modules/@pulumi/pulumi/node_modules"
cd "/Users/lukehoban/.pulumi-dev/node_modules/@pulumi/pulumi" && \
        yarn install --prefer-offline --production && \
        (yarn unlink > /dev/null 2>&1 || true) && \
        yarn link
yarn install v1.22.10
[1/5] 🔍  Validating package.json...
[2/5] 🔍  Resolving packages...
error Package "mockpackage" refers to a non-existing file '"/Users/lukehoban/.pulumi-dev/node_modules/@pulumi/pulumi/tests/mockpackage"'.
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
make: *** [install] Error 1
```

This appears to have been introduced with https://github.com/pulumi/pulumi/pull/8893.  It does not affect `make publish`, and therefore does not break the workflow used in CI, but does impact the normal local development flow.

--- a/sdk/nodejs/Makefile
+++ b/sdk/nodejs/Makefile
@@ -29,6 +29,7 @@ build_package:: ensure
 	cp tests/runtime/jsClosureCases_10_4.js bin/tests/runtime
 	cp -R tests/automation/data/. bin/tests/automation/data/
 	cp README.md ../../LICENSE ./dist/* bin/
+# safely remove mockpackage dependency see [pulumi/pulumi#9026]
 	cp package.json yarn.lock bin/ && cd bin && yarn remove mockpackage && rm yarn.lock
 	node ../../scripts/reversion.js bin/package.json ${VERSION}
 	node ../../scripts/reversion.js bin/version.js ${VERSION}

--- a/sdk/nodejs/Makefile
+++ b/sdk/nodejs/Makefile
@@ -28,11 +28,11 @@ build_package:: ensure
 	cp tests/runtime/jsClosureCases_8.js bin/tests/runtime
 	cp tests/runtime/jsClosureCases_10_4.js bin/tests/runtime
 	cp -R tests/automation/data/. bin/tests/automation/data/
-	cp README.md ../../LICENSE package.json ./dist/* bin/
+	cp README.md ../../LICENSE ./dist/* bin/
+	sed 's/"devDependencies"/"_devDependencies"/g' package.json > bin/package.json
 	node ../../scripts/reversion.js bin/package.json ${VERSION}
 	node ../../scripts/reversion.js bin/version.js ${VERSION}
 	cp -R proto/. bin/proto/
-	cp -R tests/mockpackage bin/tests/
 	mkdir -p bin/tests/runtime/langhost/cases/
 	find tests/runtime/langhost/cases/* -type d -exec cp -R {} bin/tests/runtime/langhost/cases/ \;
 

--- a/sdk/nodejs/Makefile
+++ b/sdk/nodejs/Makefile
@@ -29,7 +29,7 @@ build_package:: ensure
 	cp tests/runtime/jsClosureCases_10_4.js bin/tests/runtime
 	cp -R tests/automation/data/. bin/tests/automation/data/
 	cp README.md ../../LICENSE ./dist/* bin/
-	cat package.json | grep -v '"mockpackage"'> bin/package.json
+	cp package.json yarn.lock bin/ && cd bin && yarn remove mockpackage && rm yarn.lock
 	node ../../scripts/reversion.js bin/package.json ${VERSION}
 	node ../../scripts/reversion.js bin/version.js ${VERSION}
 	cp -R proto/. bin/proto/

--- a/sdk/nodejs/Makefile
+++ b/sdk/nodejs/Makefile
@@ -29,7 +29,7 @@ build_package:: ensure
 	cp tests/runtime/jsClosureCases_10_4.js bin/tests/runtime
 	cp -R tests/automation/data/. bin/tests/automation/data/
 	cp README.md ../../LICENSE ./dist/* bin/
-	sed 's/"devDependencies"/"_devDependencies"/g' package.json > bin/package.json
+	cat package.json | grep -v '"mockpackage"'> bin/package.json
 	node ../../scripts/reversion.js bin/package.json ${VERSION}
 	node ../../scripts/reversion.js bin/version.js ${VERSION}
 	cp -R proto/. bin/proto/

--- a/sdk/nodejs/Makefile
+++ b/sdk/nodejs/Makefile
@@ -32,6 +32,7 @@ build_package:: ensure
 	node ../../scripts/reversion.js bin/package.json ${VERSION}
 	node ../../scripts/reversion.js bin/version.js ${VERSION}
 	cp -R proto/. bin/proto/
+	cp -R tests/mockpackage bin/tests/
 	mkdir -p bin/tests/runtime/langhost/cases/
 	find tests/runtime/langhost/cases/* -type d -exec cp -R {} bin/tests/runtime/langhost/cases/ \;
 

